src/imitation/util/buffering_wrapper.py

```diff
@@ -0,0 +1,75 @@
+import numpy as np
+from stable_baselines.common.vec_env import VecEnv, VecEnvWrapper
+
+from imitation.util import rollout
+
+
+class BufferingWrapper(VecEnvWrapper):
+  """Saves transitions of underlying VecEnv.
+
+  Retrieve saved transitions using `pop_transitions()`.
+  """
+
+  def __init__(self, venv: VecEnv, error_on_premature_reset: bool = True):
+    """
+    Args:
+      venv: The wrapped VecEnv.
+      error_premature_reset: Error if `reset()` is called on this wrapper
+        and there are saved samples that haven't yet been accessed.
+    """
+    super().__init__(venv)
+    self.error_on_premature_reset = error_on_premature_reset
+    self._did_reset = False
+    self.obs_list = []
+    self.acts_list = []
+    self.rews_list = []
+    self.dones_list = []
+
+  def reset(self, **kwargs):
+    if self.error_on_premature_reset and len(self.acts_list) != 0:
+      raise RuntimeError(
+        "BufferingWrapper reset() before samples were accessed")
+    obs = self.venv.reset(**kwargs)
+    self.obs_list = [obs]
+    self.acts_list = []
+    self.rews_list = []
+    self.dones_list = []
+    self._did_reset = True
+    return obs
+
+  def step(self, actions):
```

Comment:
This will never get called if consumer uses `step_async` and `step_wait` directly.

Think should introduce `step_async` that just appends to `self.acts_list`. Then the rest of this method can be `step_wait`.